### DISCUSSION TITLE
Support for authentication proxies

### DISCUSSION THREAD

**Discussion by DevSecNinja** [outline/outline#7870] [Ideas]
It would be really great if Outline could support authentication proxies. In this case, the user is redirected by nginx to e.g. [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy) and upon successful authentication, outline can just read the `X-Forwarded-User` header to get the user ID.

Thanks!

**Comment by tommoor** [comment]
We already support Oauth2, why would a proxy be better?

**Comment by DevSecNinja** [comment]
It’s more seamless. I’m using Entra ID authentication and since my device is Entra ID joined, I can seamlessly login to applications that support authentication proxies without having to click any buttons on the page. From a security perspective, I also like to always redirect to the authentication service first before clients hit the web service to prevent potential authentication bypass vulnerabilities in the web service.
